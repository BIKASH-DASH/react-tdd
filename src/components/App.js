import React, { Component } from 'react'
import {Button}from 'react-bootstrap'

export default class App extends Component {
    constructor(){
        super();
        this.state={
            gifts:[]
        }
    }

    addGift(){
        const { gifts } = this.state ;
        gifts.push({id:gifts.length+1})
        this.setState(gifts);
    }

    render() {
        return (
            <div>
                <h2>App</h2>
                <div className='gift-list'>
                    {
                        this.state.gifts.map((gift)=>{
                            return (<div key={gift.id} >{gift.id}</div>)
                        })
                    }
                </div>
                <Button className='button-newGift' onClick={()=>this.addGift()}>Add Gift</Button>
            </div>
        )
    }
}
