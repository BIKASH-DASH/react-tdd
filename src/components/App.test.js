import React from 'react';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import App from './App';
Enzyme.configure({ adapter: new Adapter() });
const wrapper = shallow(<App/>);

describe('App components', () => {
    
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should initialize the empty list of gift', () => {
        expect(wrapper.state().gifts).toEqual([])
    });    
    
    describe('when Clicking the `Add Gift` button', () => {
        beforeEach(()=>{
            wrapper.find('.button-newGift').simulate('click');
        })

        afterEach(()=>{
            wrapper.setState({gifts:[]})
        })

        it('should add a new gift ', () => {    
            expect(wrapper.state().gifts).toEqual([{"id": 1}])
        });
    
        it('should render list ', () => {      
            expect(wrapper.find('.gift-list').children().length).toEqual(1)
        });
        
    });
    
})

