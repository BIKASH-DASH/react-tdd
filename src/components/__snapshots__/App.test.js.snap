// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components should render correctly 1`] = `ShallowWrapper {}`;
